lib/facility.rb

```diff
@@ -1,14 +1,34 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(dmv)
+    @name = dmv[:name]
+    @address = dmv[:address]
+    @phone = dmv[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0 
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless @services.include?('Vehicle Registration')
+
+    if vehicle.antique?
+      vehicle.plate_type = :antique
+      @collected_fees += 25
+    elsif vehicle.electric_vehicle?
+      vehicle.plate_type = :ev
+      @collected_fees += 200
+    else
+      vehicle.plate_type = :regular
+      @collected_fees += 100
+    end
+    vehicle.registration_date = Date.today
+    @registered_vehicles << vehicle
+    @registered_vehicles
+  end
```

Comment:
Great job with all the logic here and all the side effects we wanted this register_vehicle method to have. Something to work on moving forward, is thinking about how we can refactor this method so that this method is only doing one thing. How can we break out some of these pieces into helper methods so that each method is doing one thing?